pkg/api/upload.go

```
@@ -0,0 +1,209 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/virtual-vgo/vvgo/pkg/sheets"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type UploadType string
+
+func (x UploadType) String() string { return string(x) }
+
+const (
+	UploadTypeClix   UploadType = "clix"
+	UploadTypeSheets UploadType = "sheets"
+)
+
+type Upload struct {
+	UploadType    `json:"upload_type"`
+	*ClixUpload   `json:"clix_upload"`
+	*SheetsUpload `json:"sheets_upload"`
+	Project       string `json:"project"`
+	FileName      string `json:"file_name"`
+	FileBytes     []byte `json:"file_bytes"`
+	ContentType   string `json:"content_type"`
+}
+
+type UploadStatus struct {
+	FileName string
+	Code     int
+	Error    string
+}
+
+type ClixUpload struct {
+	PartNames   []string `json:"part_names"`
+	PartNumbers []int    `json:"part_numbers"`
+}
+
+type SheetsUpload struct {
+	PartNames   []string `json:"part_names"`
+	PartNumbers []int    `json:"part_numbers"`
+}
+
+const UploadsTimeout = 10 * time.Second
+
+func (x *Server) Upload(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		methodNotAllowed(w)
+		return
+	}
+
+	if r.Header.Get("Content-Type") != "application/json" {
+		invalidContent(w)
+		return
+	}
+
+	var documents []Upload
+	if ok := jsonDecode(r.Body, &documents); !ok {
+		badRequest(w, "")
+		return
+	}
+
+	if len(documents) == 0 {
+		return // nothing to do
```

Probably want to return an error response here